src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -48,12 +52,20 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
             var diagnosticListenerEnabled = _diagnosticListener.IsEnabled();
             var loggingEnabled = _logger.IsEnabled(LogLevel.Critical);
 
+            // TODO: If loggingEnabled check to change to if ActivityPropogation options are enabled
             if (loggingEnabled || (diagnosticListenerEnabled && _diagnosticListener.IsEnabled(ActivityName, httpContext)))
             {
-                context.Activity = StartActivity(httpContext, out var hasDiagnosticListener);
-                context.HasDiagnosticListener = hasDiagnosticListener;
+                _dummyListener = new ActivityListener()
+                {
+                    ShouldListenTo = activitySource => activitySource.Name.Equals(ActivitySourceName),
```

Comment:
>activitySource.Name.Equals(ActivitySourceName) [](start = 55, length = 46)

I would use Object.ReferenceEquals(activitySource, _activitySource) here instead. That ensure avoid listening to any other ActivitySource can be created with the same name.